sections/user/take_edit.php

```diff
@@ -153,6 +153,74 @@
 // End building $Paranoia
 
 
+// API Key Helpers
+
+function doesUserHasToken(int $UserID, $DB): bool {
+    return $DB->scalar("
+        SELECT 1
+        FROM api_applications
+        WHERE UserID = $UserID"
+    ) === 1;
+}
+
+function hasApiToken(int $userId, string $token, $DB): bool {
+    return $DB->scalar("
+        SELECT 1
+        FROM api_applications
+        WHERE UserID = $userId
+            AND Token = '$token'"
+    ) === 1;
+}
+
+function revokeApiTokenById(int $UserID, $DB): int {
+    $DB->prepared_query("
+        DELETE FROM api_applications
+        WHERE UserID = ? ", $UserID
+    );
+    return $DB->affected_rows();
+}
+
+function createApiToken(int $UserID, string $key, $DB): string {
+    $suffix = sprintf('%014d', $UserID);
+    $name = "API_TOKEN";
+
+    while (true) {
+        // prevent collisions with an existing token name
+        $token = base64UrlEncode(encrypt(random_bytes(32) . $suffix, $key));
+        if (!hasApiToken($UserID, $token, $DB))
+            break;
+    }
+
+    $DB->prepared_query("
+        INSERT INTO api_applications
+               (UserID, Name, Token)
+        VALUES (?,       ?,    ?)", $UserID, $name, $token
+    );
+    return $token;
+}
+
+function encrypt($plaintext, $key): string {
```

Comment:
1. It's similar as `app/Util/Crypto.php` encrypt function.
2. Why use so complicate encrypt?, Why not use md5

Reply:
This is the original code from opusnet codebase

Reply:
so you can use `app/Util/Crypto.php`

Reply:
done